integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+}
+
+
+Cypress.Commands.add("apiCheckDevices", (userName: string, expectedDevices: [ComparableDevice], params: string = {},  statusCode: number = 200) => {
+  const fullUrl = v1ApiPath('devices',params);
+
+  logTestDescription(
+      `${userName} Check devices seen by  user '${userName}'`,
+      { user: userName},
+      true
+  );
+
+  makeAuthorizedRequestWithStatus(
+      {
+        method: "GET",
+        url: fullUrl,
+        body: null
+      },
+      userName,
+      statusCode
+    ).then((response)=>{
+       if(statusCode==null || statusCode==200) {
+	  var devices=response.body.devices.rows;
+	  //TODO: Issue 63.  Reenable this when devices count is correct
+          //expect(response.body.devices.count).to.equal(expectedDevices.length);
+          expect(devices.length).to.equal(expectedDevices.length);
+	  var dev_count;
+	  var user_count;
+          var sortDevices=sortArrayOn(devices,'devicename');
+          var sortExpectedDevices=sortArrayOn(expectedDevices,'devicename');
+          for (dev_count=0; dev_count < expectedDevices.length; dev_count++) {
+  	    expect(sortDevices[dev_count].id).to.equal(sortExpectedDevices[dev_count].id);
+    	    expect(sortDevices[dev_count].devicename).to.equal(sortExpectedDevices[dev_count].devicename);
+	    expect(sortDevices[dev_count].active).to.equal(sortExpectedDevices[dev_count].active);
+            if(sortExpectedDevices[dev_count].Users==null) {
+	      expect(sortDevices[dev_count].Users).to.not.exist;
+            } else {
+              expect(sortDevices[dev_count].Users.length).to.equal(sortExpectedDevices[dev_count].Users.length);
+	      // sort users and expected users to ensure order is the same
+              var users=sortArrayOn(sortDevices[dev_count].Users,'username');
+              var expectedUsers=sortArrayOn(sortExpectedDevices[dev_count].Users,'username');
+
+	      // compare user list
+	      var count
+              for (count=0; count < expectedUsers.length; count++) {
+  		  expect(users[count].username).to.equal(expectedUsers[count].username);
+  		  expect(users[count].id).to.equal(expectedUsers[count].id);
+  		  expect(users[count].DeviceUsers.admin).to.equal(expectedUsers[count].DeviceUsers.admin);
+  		  expect(users[count].DeviceUsers.UserId).to.equal(expectedUsers[count].DeviceUsers.UserId);
+	      };
+  	    };
+    	  }
+       };
+  });
+});
+
+Cypress.Commands.add("apiCheckDevices_contains", (userName: string, expectedDevices: [ComparableDevice], params: string = {}, statusCode: number = 200) => {
+  const fullUrl = v1ApiPath('devices',params);
+  logTestDescription(
+      `${userName} Check devices seen by user '${userName}' contain the expected devices `,
+      { user: userName  },
+      true
+  );
+
+  makeAuthorizedRequestWithStatus(
+      {
+        method: "GET",
+        url: fullUrl,
+        body: null
+      },
+      userName,
+      statusCode
+    ).then((response)=>{
+       if(statusCode==null || statusCode==200) {
```

Comment:
indent properly, alsl looks like some very similar parts to above can this be put into smaller chunks of re usable code?

Reply:
Yes.  Laziness! Extracted to function & reused.